src/main/java/dk/kb/util/yaml/YAML.java

```diff
@@ -1008,8 +1028,17 @@ private static void mergeValueToYAML(String path, LinkedHashMap<Object, Object>
         }
         base.put(key, mergeEntry(path + "." + key, base.get(key), value, defaultMA, listMA));
     }
-
-
+    
+    public boolean extrapolateSystemProperties() {
+        return extrapolateSystemProperties;
+    }
+    
+    public YAML extrapolateSystemProperties(boolean extrapolateSystemProperties) {
```

Comment:
Should probably be `doesExtrapolateSystemProperties()` and `setExtrapolateProperties(boolean)` or just `doesExtrapolate` og `setExtrapolate(boolean)`, where the *javadoc* (cough cough missing cough) explains what happens.

...Or maybe it should follow the new style and be `extrapolates()` and `extrapolate(boolean)` as the `extrapolate(boolean)` works as part of a construction chain?

Reply:
```java
  /**
     * If the YAML will extrapolate the current values of System.getProperties() in the values returned
     * @return true if extrapolation is enabled
     */
    public boolean extrapolateSystemProperties() {
        return extrapolateSystemProperties;
    }
    
    /**
     * Set the YAML to extrapolate the current values of System.getProperties() in the values returned
     * @return this YAML, not a copy
     */
    public YAML extrapolateSystemProperties(boolean extrapolateSystemProperties) {
        this.extrapolateSystemProperties = extrapolateSystemProperties;
        return this;
    }
    
    /**
     * If the YAML will extrapolate the current values of System.getProperties() in the values returned
     * @return true if extrapolation is enabled
     */
    public boolean isExtrapolateSystemProperties() {
        return extrapolateSystemProperties;
    }
    
    /**
     * Set the YAML to extrapolate the current values of System.getProperties() in the values returned
     */
    public void setExtrapolateSystemProperties(boolean extrapolateSystemProperties) {
        this.extrapolateSystemProperties = extrapolateSystemProperties;
    }
```

Reply:
Is it a "mock Toke" or a misunderstanding that you made both `isExtrapolateSystemProperties` og `extrapolateSystemProperties`? :-)

Anyways, it it still not clear from the JavaDoc what is going on. **I** know, but I havde played with it before. What about "If extrapolation is enabled, system property values can be used as YAML values, such as {@code greeting: "Running under ${user.name}"} or {@code threads: ${myapp.threads}}. Custom properties can be specified for the application from command line {@code java -Dmyapp.threads=4 -jar myapp.jar}, container configuration or similar.